### PR TITLE
CSSRE-2073: Removal of login form options

### PR DESCRIPTION
**Pull request by vinaybommana**
from frontend

#### What:
* added the frontend changes for removal of login form options using environment variables

#### Why:
(from the ticket)
As a service owner, I want to be able to ensure that only the supported login form options are presented to a user so that;

1. I can minimize the possible security-related threats (attack vectors) related to those unsupported login form options.
2. Provide a simpler and correct customer experience.

#### Tickets:
https://issues.redhat.com/browse/CSSRE-2073

### CODE REVIEW

**Comment by vinaybommana** [comment]
Hi @Nanyte25 please take a look at this when you have some time..

File: openshift/template.yaml [hunk 1]
```diff
@@ -436,6 +438,11 @@ parameters:
     name: ENABLE_SOCIAL_AUTH
     value: "True"
 
+  - description: ENABLE_LOGIN_FORM
+    displayName: ENABLE_LOGIN_FORM
+    name: ENABLE_LOGIN_FORM
```
Comment (Nanyte25) [inline openshift/template.yaml:443]: @vinaybommana Great work, so this flag `ENABLE_LOGIN_FORM
    value: "False"` will enable or disable the the login form via the deployment template?

File: src/app/login/login.component.html [hunk 2]
```diff
@@ -63,8 +66,9 @@ <h1 i18n>Login</h1>
             >Log in</button
           >
         </form>
+      </div>
         <ng-container *ngIf="socialApps$ | async as socialApps">
-          <div *ngIf="socialApps.length" class="or">
+          <div *ngIf="socialApps.length > 1" class="or">
```
Comment (Nanyte25) [inline src/app/login/login.component.html:71]: IS this the correct way to do this  in Angular `<div *ngIf="socialApps.length > 1" class="or">`?

Comment (Nanyte25) [reply src/app/login/login.component.html:71]: @vinaybommana is there somewhere I can see this working?

**Review by Nanyte25 — CHANGES_REQUESTED** [review]

**Comment by vinaybommana** [comment]
Hi @Nanyte25, so the enabling and disabling consists of two changes, the `ENABLE_LOGIN_FORM` if "False" disables the backend login urls, so that login won't be possible through rest-api.

the frontend cosmetic change mostly depends on `environment.ts` which has flag: `loginForm`, if `false` disables the login from frontend.

```html
<div *ngIf="socialApps.length > 1" class="or">
```
so this is regarding the **"or"**
Screenshot_20220713_151226

between manual login form and "github" login, so I've added the condition to add *or* if LoginForm or multilple *socialApps* exists

**Review by Nanyte25 — COMMENTED** [review]
@vinaybommana  I am seeing the login form disable and the /admin end-point is accessible still. However are s you sure you have github auth working correctly? its need to point to a default `org` in the `site path url`. This can be configure in Django-admin. If you need to test this
Screenshot 2022-07-13 at 12 35 35

**Comment by vinaybommana** [comment]
Hi @Nanyte25 , the github auth is working basically the `site` inorder to work I've given here as
Screenshot_20220713_171800

and the github social application as 
Screenshot_20220713_171852

the social url from `DOMAIN` is basically retrieved from `openshift/template.yml` since we are port-forwarding to local machine we can't successfully login. let me change the `social.py` for testing purpose. 
we actually don't need to change that 
```
-        return DOMAIN + "/auth/" + provider_id
+        # return DOMAIN + "/auth/" + provider_id
+        return "http://127.0.0.1:8080" + "/auth/" + provider_id
```

**Review by Nanyte25 — APPROVED** [review]

**Pull request merged**